CRM/Core/Payment/Dummy.php

```diff
@@ -107,26 +94,25 @@ public function doDirectPayment(&$params) {
       $result['trxn_id'] = array_shift($this->_doDirectPaymentResult['trxn_id']);
       return $result;
     }
-    if ($this->_mode == 'test') {
+    if ($this->_mode === 'test') {
       $query = "SELECT MAX(trxn_id) FROM civicrm_contribution WHERE trxn_id LIKE 'test\\_%'";
       $p = [];
-      $trxn_id = strval(CRM_Core_DAO::singleValueQuery($query, $p));
+      $trxn_id = (string) CRM_Core_DAO::singleValueQuery($query, $p);
       $trxn_id = str_replace('test_', '', $trxn_id);
-      $trxn_id = intval($trxn_id) + 1;
+      $trxn_id = (int) $trxn_id + 1;
       $params['trxn_id'] = 'test_' . $trxn_id . '_' . uniqid();
     }
     else {
       $query = "SELECT MAX(trxn_id) FROM civicrm_contribution WHERE trxn_id LIKE 'live_%'";
       $p = [];
-      $trxn_id = strval(CRM_Core_DAO::singleValueQuery($query, $p));
+      $trxn_id = (string) CRM_Core_DAO::singleValueQuery($query, $p);
       $trxn_id = str_replace('live_', '', $trxn_id);
-      $trxn_id = intval($trxn_id) + 1;
+      $trxn_id = (int) $trxn_id + 1;
       $params['trxn_id'] = 'live_' . $trxn_id . '_' . uniqid();
     }
-    $params['gross_amount'] = $params['amount'];
+    $params['gross_amount'] = $propertyBag->getAmount();
     // Add a fee_amount so we can make sure fees are handled properly in underlying classes.
```

Comment:
We don't use `gross_amount` - it should be `total_amount`

Reply:
Yeah - we can visit after as to why this is being done - that's the return stuff